src/Compilers/CSharp/Portable/Binder/ForEachLoopBinder.cs

```diff
@@ -722,15 +722,16 @@ private ForEachEnumeratorInfo.Builder GetDefaultEnumeratorInfo(ForEachEnumerator
 
             if (collectionExprType.IsDynamic())
             {
-                builder.ElementType = ((_syntax as ForEachStatementSyntax)?.Type.IsVar == true) ?
+                builder.ElementType = TypeSymbolWithAnnotations.Create(((_syntax as ForEachStatementSyntax)?.Type.IsVar == true) ?
                     (TypeSymbol)DynamicTypeSymbol.Instance :
-                    GetSpecialType(SpecialType.System_Object, diagnostics, _syntax);
+                    GetSpecialType(SpecialType.System_Object, diagnostics, _syntax),
```

Comment:
nit: with the new parameter, it would be good indent the ternary lines to make it a bit more readable.